### DISCUSSION TITLE
Reflect non-static borrows with `better_any`

### DISCUSSION THREAD

**Discussion by tancop** [bevyengine/bevy#8482] [Ideas]
[better_any](https://docs.rs/better_any/latest/better_any/) is a crate that lets you get the `TypeId` for types that don't implement `Any` because they contain non-static borrowed data. This only works on types with a single lifetime, but that's enough for most cases (even things like iterators). Using this crate would partially remove the most important limitation of `bevy_reflect` that all types have to be `'static`.

**Comment by MrGVSV** [comment]
This is an interesting project, and could be something we look into in the future (or even roll out our own version of it built-in to the `Reflect` derive macro).

I have some hesitance with using something that the Rust team is not currently making efforts to support. We risk something breaking because Rust won't support it (I haven't really dived deep into the internals of this crate so maybe it's generally safe).

And it only working with a single lifetime is also not ideal obviously. Along with the fact that `Any` is a much more standard API than `Tid`.
